torchbenchmark/score/compute_score_v1.py

```diff
@@ -0,0 +1,193 @@
+            eager_name = left + "-eager" + right
+            # We assume if a jit test exists, there must be an eager test
+            assert eager_name in norm['benchmarks'], f"Can't find eager test name {eager_test_name}"
+            result_ref['benchmarks'][jit_name] = dict()
+            result_ref['benchmarks'][jit_name]['jit_norm'] = norm['benchmarks'][jit_name]['norm']
+            result_ref['benchmarks'][jit_name]['eager_norm'] = norm['benchmarks'][eager_name]['norm']
+        return result_ref
+
+    def _test_to_config(self, name):
+        if "-freeze" in name:
+            name.replace("-freeze", "", 1)
+        test, model_name, device, mode = re.match(r"test_(.*)\[(.*)\-(.*)\-(.*)\]", name).groups()
+        return (test, model_name, device, mode)
+
+    def _config_to_weight(self, config):
+        # config weight rule in V1: 1x CPU Training, 2x GPU Training, 2x CPU Inference, 2x GPU Inference
+        test, model_name, device, mode = config
+        if test == "train" and device == "cpu":
+            return 1.0
+        else:
+            return 2.0
+        
+    # Generate the domain weights from the ref object
+    def _setup_weights(self, ref):
+        # Setup domain_weights
+        domain_weights = defaultdict(float)
+        config_weights = defaultdict(float)
+        config_dict = defaultdict(lambda: defaultdict(int))
+        task_dict = defaultdict(lambda: defaultdict(set))
+        name_dict = defaultdict(lambda: defaultdict(Enum))
+        for name in ref['benchmarks']:
+            config = self._test_to_config(name)
+            task = _get_model_task(config[1])
+            # print(f"Found domain {type(task).__name__}, task: {task.name} in model {name}")
+            task_dict[type(task).__name__][task.name].add(config[1])
+            config_dict[config[1]][name] = self._config_to_weight(config)
+            name_dict[name] = task
+        category_cnt = len(task_dict)
+        sum_domain = 0.0
+        for name in name_dict:
+            task = name_dict[name]
+            domain_cnt = len(task_dict[type(task).__name__])
+            task_cnt = len(task_dict[type(task).__name__][task.name])
+            domain_weights[name] = (1.0 / category_cnt) * (1.0 / domain_cnt) * (1.0 / task_cnt)
+        # Setup config_weights
+        for name in name_dict:
+            (test, model_name, device, mode) = self._test_to_config(name)
+            config_weights[name] = config_dict[model_name][name] / sum(config_dict[model_name].values())
+        for name in name_dict:
+            total_weight = config_weights[name] * domain_weights[name]
+            sum_domain += total_weight
+        assert(abs(sum_domain - 1.0) < 1e-6), "The total task weights sum is not 1.0, please submit a bug report."
+        return (domain_weights, config_weights)
+ 
+    def _setup_benchmark_norms(self, ref_data):
+        """
+        Helper function which gets the normalization values per benchmark
```

Comment:
Is normalization = speedup over reference value?

Reply:
Here, the norm is only the absolute latency number in seconds. However, the relative score is the weighted average of the speedup over reference value: https://github.com/pytorch/benchmark/pull/401/files#diff-64db27bada97956276b540db8c8c4056a13f748f27508e2608fae8b3be60c7b5R133